File: Dummy/hawkshaw/library/ktextensions/ContextKt.java
package com.hawkshaw.library.ktextensions;

import E.k;
import Y1.K;
import android.app.Activity;
import android.app.ActivityManager;
import android.app.AppOpsManager;
import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.content.ContextWrapper;
import android.os.Build;
import android.os.Process;
import android.widget.Toast;
import com.hawkshaw.library.logger.Logger;
import i.C0528x;
import java.util.Iterator;
import java.util.List;

public final class ContextKt {
    /* JADX WARNING: Code restructure failed: missing block: B:7:0x0023, code lost:*/
        r2 = r2.getNotificationChannel(r3);
     
    /* Code decompiled incorrectly, please refer to instructions dump. */
    public static final boolean areNotificationsEnabled(android.content.Context r2, java.lang.String r3) {
        
            java.lang.String r0 = "<this>"
            Y1.K.n(r2, r0)
            r0 = 0
            if (r3 == 0) goto L_0x0030
            boolean r1 = r3.j.U(r3)
            if (r1 == 0) goto L_0x000f
            goto L_0x0030
        L_0x000f:
            java.lang.String r1 = "notification"
            java.lang.Object r2 = r2.getSystemService(r1)
            java.lang.String r1 = "null cannot be cast to non-null type android.app.NotificationManager"
            Y1.K.l(r2, r1)
            android.app.NotificationManager r2 = (android.app.NotificationManager) r2
            boolean r1 = r2.areNotificationsEnabled()
            if (r1 != 0) goto L_0x0023
            return r0
        L_0x0023:
            android.app.NotificationChannel r2 = r2.getNotificationChannel(r3)
            if (r2 == 0) goto L_0x0030
            int r2 = r2.getImportance()
            if (r2 == 0) goto L_0x0030
            r0 = 1
        L_0x0030:
            return r0
        
        throw new UnsupportedOperationException("Method not decompiled: com.hawkshaw.library.ktextensions.ContextKt.areNotificationsEnabled(android.content.Context, java.lang.String):boolean");
    }

    public static final boolean canLaunchFromBg(Context context) {
        K.n(context, "<this>");
        return Build.VERSION.SDK_INT < 29 || isAppInForeground(context);
    }

    public static final void copyToClipboard(Context context, CharSequence charSequence, String str) {
        K.n(context, "<this>");
        K.n(charSequence, "text");
        K.n(str, "label");
        ClipboardManager clipboardManager = (ClipboardManager) k.getSystemService(context, ClipboardManager.class);
        if (clipboardManager != null) {
            clipboardManager.setPrimaryClip(ClipData.newPlainText(str, charSequence));
        }
    }

    public static /* synthetic */ void copyToClipboard$default(Context context, CharSequence charSequence, String str, int i5, Object obj) {
        if ((i5 & 2) != 0) {
            str = "label";
        }
        copyToClipboard(context, charSequence, str);
    }

    public static final Activity getActivity(Context context) {
        K.n(context, "<this>");
        if (context instanceof Activity) {
            return (Activity) context;
        }
        if (context instanceof ContextWrapper) {
            Context baseContext = ((ContextWrapper) context).getBaseContext();
            K.m(baseContext, "getBaseContext(...)");
            return getActivity(baseContext);
        }
        throw new Exception("Activity not found");
    }

    public static final boolean isAppInForeground(Context context) {
        Integer num;
        Object obj;
        K.n(context, "<this>");
        Context applicationContext = context.getApplicationContext();
        Object systemService = context.getSystemService("activity");
        K.l(systemService, "null cannot be cast to non-null type android.app.ActivityManager");
        List<ActivityManager.RunningAppProcessInfo> runningAppProcesses = ((ActivityManager) systemService).getRunningAppProcesses();
        if (runningAppProcesses == null) {
            return false;
        }
        Iterator it = runningAppProcesses.iterator();
        while (true) {
            num = null;
            if (!it.hasNext()) {
                obj = null;
                break;
            }
            obj = it.next();
            if (K.f(((ActivityManager.RunningAppProcessInfo) obj).processName, applicationContext.getPackageName())) {
                break;
            }
        }
        ActivityManager.RunningAppProcessInfo runningAppProcessInfo = (ActivityManager.RunningAppProcessInfo) obj;
        ActivityManager.getMyMemoryState(runningAppProcessInfo);
        Logger logger = Logger.INSTANCE;
        if (runningAppProcessInfo != null) {
            num = Integer.valueOf(runningAppProcessInfo.importance);
        }
        Logger.v$default(logger, "IsAppInForeground", "Importance: " + num, false, 4, (Object) null);
        return runningAppProcessInfo != null && runningAppProcessInfo.importance == 100;
    }

    public static final boolean isAudioRecordAllowed(Context context) {
        K.n(context, "<this>");
        if (Build.VERSION.SDK_INT < 29) {
            return true;
        }
        Object systemService = context.getSystemService("appops");
        K.l(systemService, "null cannot be cast to non-null type android.app.AppOpsManager");
        int a5 = ((AppOpsManager) systemService).unsafeCheckOpNoThrow("android:record_audio", Process.myUid(), context.getPackageName());
        Logger.v$default(Logger.INSTANCE, "AppOpsManager", C0528x.b("isAudioRecordAllowed: ", a5), false, 4, (Object) null);
        return a5 == 0;
    }

    public static final boolean isCameraAllowed(Context context) {
        K.n(context, "<this>");
        if (Build.VERSION.SDK_INT < 29) {
            return true;
        }
        Object systemService = context.getSystemService("appops");
        K.l(systemService, "null cannot be cast to non-null type android.app.AppOpsManager");
        int o4 = ((AppOpsManager) systemService).unsafeCheckOpNoThrow("android:camera", Process.myUid(), context.getPackageName());
        Logger.v$default(Logger.INSTANCE, "AppOpsManager", C0528x.b("isCameraAllowed: ", o4), false, 4, (Object) null);
        return o4 == 0;
    }

    /* JADX DEBUG: Multi-variable search result rejected for TypeSearchVarInfo{r1v1, resolved type: X2.q} */
    /* JADX DEBUG: Multi-variable search result rejected for TypeSearchVarInfo{r1v2, resolved type: java.util.ArrayList} */
    /* JADX DEBUG: Multi-variable search result rejected for TypeSearchVarInfo{r1v3, resolved type: X2.q} */
    /* JADX DEBUG: Multi-variable search result rejected for TypeSearchVarInfo{r1v4, resolved type: X2.q} */
    /* JADX WARNING: Multi-variable type inference failed */
    /* Code decompiled incorrectly, please refer to instructions dump. */
    public static final java.util.List<java.lang.String> readFromClipboard(android.content.Context r5) {
        
            java.lang.String r0 = "<this>"
            Y1.K.n(r5, r0)
            java.lang.Class<android.content.ClipboardManager> r0 = android.content.ClipboardManager.class
            java.lang.Object r0 = E.k.getSystemService(r5, r0)
            android.content.ClipboardManager r0 = (android.content.ClipboardManager) r0
            X2.q r1 = X2.q.f2602D
            if (r0 != 0) goto L_0x0012
            return r1
        L_0x0012:
            android.content.ClipData r0 = r0.getPrimaryClip()
            if (r0 == 0) goto L_0x0032
            java.util.ArrayList r1 = new java.util.ArrayList
            r1.<init>()
            int r2 = r0.getItemCount()
            r3 = 0
        L_0x0022:
            if (r3 >= r2) goto L_0x0032
            android.content.ClipData$Item r4 = r0.getItemAt(r3)
            java.lang.String r4 = r4.coerceToHtmlText(r5)
            r1.add(r4)
            int r3 = r3 + 1
            goto L_0x0022
        L_0x0032:
            return r1
        
        throw new UnsupportedOperationException("Method not decompiled: com.hawkshaw.library.ktextensions.ContextKt.readFromClipboard(android.content.Context):java.util.List");
    }

    public static final void toast(Context context, CharSequence charSequence) {
        K.n(context, "<this>");
        K.n(charSequence, "text");
        Toast.makeText(context, charSequence, 0).show();
    }

    public static final void toastLong(Context context, CharSequence charSequence) {
        K.n(context, "<this>");
        K.n(charSequence, "text");
        Toast.makeText(context, charSequence, 1).show();
    }
}

File: app/src/main/java/com/hawkshaw/library/ktextensions/ContextKt.kt
package com.hawkshaw.library.ktextensions

import android.app.Activity
import android.app.ActivityManager
import android.app.AppOpsManager
import android.app.NotificationManager
import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.content.ContextWrapper
import android.os.Build
import android.os.Process
import android.widget.Toast
import androidx.annotation.RequiresApi
import com.hawkshaw.library.logger.Logger
import java.util.ArrayList
import android.util.Log // Added for logging

object ContextKt {
    private const val TAG = "ContextKtExt" // Added for logging

    @RequiresApi(Build.VERSION_CODES.O)
    @JvmStatic
    fun areNotificationsEnabled(context: Context, channelId: String?): Boolean {
        Log.d(TAG, "[DEBUG] areNotificationsEnabled called. Context: $context, ChannelId: $channelId")
        if (channelId.isNullOrEmpty()) {
            Log.d(TAG, "[DEBUG] areNotificationsEnabled: ChannelId is null or empty. Returning false.")
            return false
        }

        val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        Log.d(TAG, "[DEBUG] areNotificationsEnabled: Got NotificationManager.")
        
        val globalNotificationsEnabled = notificationManager.areNotificationsEnabled()
        Log.d(TAG, "[DEBUG] areNotificationsEnabled: Global notifications enabled: $globalNotificationsEnabled")
        if (!globalNotificationsEnabled) {
            Log.d(TAG, "[DEBUG] areNotificationsEnabled: Global notifications are disabled. Returning false.")
            return false
        }

        val channel = notificationManager.getNotificationChannel(channelId)
        Log.d(TAG, "[DEBUG] areNotificationsEnabled: NotificationChannel for ID '$channelId': $channel")
        
        val result = channel != null && channel.importance != NotificationManager.IMPORTANCE_NONE
        Log.d(TAG, "[DEBUG] areNotificationsEnabled: Channel exists and importance is not NONE: $result. Channel importance: ${channel?.importance}")
        
        return result 
    }

    @JvmStatic
    fun canLaunchFromBg(context: Context): Boolean {
        Log.d(TAG, "[DEBUG] canLaunchFromBg called. Context: $context")
        val sdkInt = Build.VERSION.SDK_INT
        Log.d(TAG, "[DEBUG] canLaunchFromBg: SDK_INT: $sdkInt (Q is ${Build.VERSION_CODES.Q})")
        if (sdkInt < Build.VERSION_CODES.Q) {
            Log.d(TAG, "[DEBUG] canLaunchFromBg: SDK < Q. Returning true.")
            return true
        }
        val appInForeground = isAppInForeground(context) // isAppInForeground already logs
        Log.d(TAG, "[DEBUG] canLaunchFromBg: SDK >= Q. isAppInForeground result: $appInForeground. Returning $appInForeground.")
        return appInForeground
    }

    @JvmStatic
    fun copyToClipboard(context: Context, text: CharSequence, label: String = "label") {
        Log.d(TAG, "[DEBUG] copyToClipboard called. Context: $context, Text: \"$text\", Label: \"$label\"")
        val clipboardManager = context.getSystemService(Context.CLIPBOARD_SERVICE) as? ClipboardManager
        if (clipboardManager != null) {
            clipboardManager.setPrimaryClip(ClipData.newPlainText(label, text))
            Log.d(TAG, "[DEBUG] copyToClipboard: Text copied to clipboard.")
        } else {
            Log.w(TAG, "[DEBUG] copyToClipboard: ClipboardManager is null.")
        }
    }

    @JvmStatic
    fun getActivity(context: Context): Activity {
        Log.d(TAG, "[DEBUG] getActivity called. Current context: $context (Type: ${context.javaClass.name})")
        return when (context) {
            is Activity -> {
                Log.d(TAG, "[DEBUG] getActivity: Context is Activity. Returning context.")
                context
            }
            is ContextWrapper -> {
                Log.d(TAG, "[DEBUG] getActivity: Context is ContextWrapper. Calling getActivity on baseContext: ${context.baseContext}")
                getActivity(context.baseContext) // Recursive call
            }
            else -> {
                Log.e(TAG, "[DEBUG] getActivity: Activity not found in context hierarchy. Throwing exception.")
                throw Exception("Activity not found")
            }
        }
    }

    @JvmStatic
    fun isAppInForeground(context: Context): Boolean {
        Log.d(TAG, "[DEBUG] isAppInForeground called. Context: $context")
        val applicationContext = context.applicationContext
        Log.d(TAG, "[DEBUG] isAppInForeground: ApplicationContext: $applicationContext, PackageName: ${applicationContext.packageName}")
        val activityManager = context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
        val runningProcesses = activityManager.runningAppProcesses
        
        if (runningProcesses == null) {
            Log.w(TAG, "[DEBUG] isAppInForeground: runningAppProcesses is null. Returning false.")
            return false
        }
        Log.d(TAG, "[DEBUG] isAppInForeground: Found ${runningProcesses.size} running processes.")

        val processInfo = runningProcesses.find {
            it.processName == applicationContext.packageName
        }
        
        if (processInfo == null) {
            Log.w(TAG, "[DEBUG] isAppInForeground: ProcessInfo for package ${applicationContext.packageName} not found. Returning false.")
            return false
        }
        Log.d(TAG, "[DEBUG] isAppInForeground: Found ProcessInfo: $processInfo")

        val result = processInfo.importance == ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND
        Log.d(TAG, "[DEBUG] isAppInForeground: Process importance: ${processInfo.importance}. Is foreground: $result.")
        return result
    }

    @RequiresApi(Build.VERSION_CODES.Q)
    @JvmStatic
    fun isAudioRecordAllowed(context: Context): Boolean {
        Log.d(TAG, "[DEBUG] isAudioRecordAllowed called. Context: $context")
        val sdkInt = Build.VERSION.SDK_INT
        Log.d(TAG, "[DEBUG] isAudioRecordAllowed: SDK_INT: $sdkInt (Q is ${Build.VERSION_CODES.Q})")
        if (sdkInt < Build.VERSION_CODES.Q) {
            Log.d(TAG, "[DEBUG] isAudioRecordAllowed: SDK < Q. Returning true.")
            return true
        }

        val appOpsManager = context.getSystemService(Context.APP_OPS_SERVICE) as AppOpsManager
        Log.d(TAG, "[DEBUG] isAudioRecordAllowed: Got AppOpsManager.")
        val mode = appOpsManager.unsafeCheckOpNoThrow(
            "android:record_audio",
            Process.myUid(),
            context.packageName
        )
        Log.d(TAG, "[DEBUG] isAudioRecordAllowed: AppOps mode for 'android:record_audio': $mode (MODE_ALLOWED is ${AppOpsManager.MODE_ALLOWED})")
        
        val result = mode == AppOpsManager.MODE_ALLOWED
        Log.d(TAG, "[DEBUG] isAudioRecordAllowed: Returning $result.")
        return result
    }

    @RequiresApi(Build.VERSION_CODES.Q)
    @JvmStatic
    fun isCameraAllowed(context: Context): Boolean {
        Log.d(TAG, "[DEBUG] isCameraAllowed called. Context: $context")
        val sdkInt = Build.VERSION.SDK_INT
        Log.d(TAG, "[DEBUG] isCameraAllowed: SDK_INT: $sdkInt (Q is ${Build.VERSION_CODES.Q})")
        if (sdkInt < Build.VERSION_CODES.Q) {
            Log.d(TAG, "[DEBUG] isCameraAllowed: SDK < Q. Returning true.")
            return true
        }

        val appOpsManager = context.getSystemService(Context.APP_OPS_SERVICE) as AppOpsManager
        Log.d(TAG, "[DEBUG] isCameraAllowed: Got AppOpsManager.")
        val mode = appOpsManager.unsafeCheckOpNoThrow(
            "android:camera",
            Process.myUid(),
            context.packageName
        )
        Logger.v("AppOpsManager", "isCameraAllowed: $mode", false, 4, null)
        Log.d(TAG, "[DEBUG] isCameraAllowed: AppOps mode for 'android:camera': $mode (MODE_ALLOWED is ${AppOpsManager.MODE_ALLOWED})")
        
        val result = mode == AppOpsManager.MODE_ALLOWED
        Log.d(TAG, "[DEBUG] isCameraAllowed: Returning $result.")
        return result
    }

    @JvmStatic
    fun readFromClipboard(context: Context): List<String> {
        Log.d(TAG, "[DEBUG] readFromClipboard called. Context: $context")
        val clipboardManager = context.getSystemService(Context.CLIPBOARD_SERVICE) as? ClipboardManager
        if (clipboardManager == null) {
            Log.w(TAG, "[DEBUG] readFromClipboard: ClipboardManager is null. Returning empty list.")
            return emptyList()
        }
        
        val clipData = clipboardManager.primaryClip
        if (clipData == null) {
            Log.d(TAG, "[DEBUG] readFromClipboard: Primary clip is null. Returning empty list.")
            return emptyList()
        }
        Log.d(TAG, "[DEBUG] readFromClipboard: Got primary clip. Item count: ${clipData.itemCount}")

        val result = ArrayList<String>()
        for (i in 0 until clipData.itemCount) {
            val itemText = clipData.getItemAt(i).coerceToHtmlText(context)
            Log.d(TAG, "[DEBUG] readFromClipboard: Item $i text: \"${itemText}\"")
            result.add(itemText)
        }
        Log.d(TAG, "[DEBUG] readFromClipboard: Returning list with ${result.size} items.")
        return result
    }

    @JvmStatic
    fun toast(context: Context, text: CharSequence) {
        Log.d(TAG, "[DEBUG] toast (short) called. Context: $context, Text: \"$text\"")
        Toast.makeText(context, text, Toast.LENGTH_SHORT).show()
    }

    @JvmStatic
    fun toastLong(context: Context, text: CharSequence) {
        Log.d(TAG, "[DEBUG] toastLong called. Context: $context, Text: \"$text\"")
        Toast.makeText(context, text, Toast.LENGTH_LONG).show()
    }
}
